src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -49,9 +49,8 @@ public override void CopyFile(string sourceFullPath, string destFullPath, bool o
             // has varying degrees of support on different systems.
 
             // Copy the contents of the file from the source to the destination, creating the destination in the process
-            const int bufferSize = 0x1000;
-            using (UnixFileStream src = new UnixFileStream(sourceFullPath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize, FileOptions.None))
-            using (UnixFileStream dst = new UnixFileStream(destFullPath, overwrite ? FileMode.CreateNew : FileMode.Create, FileAccess.Read, FileShare.Read, bufferSize, FileOptions.None))
+            using (Stream src = File.OpenRead(sourceFullPath))
+            using (Stream dst = File.OpenWrite(destFullPath))
```

Comment:
This ignores the overwrite arg.  You need to throw when overwrite is false and the file exists.  That is what FileMode.CreateNew was doing for you before.

Instead this should be File.Open(destFullPath, overwrite ? FileMode.CreateNew : FileMode.Create)

PS: previously the code was attempting to create a file with only read access, that should have never worked.  If it did there may be a bug in UnixFileStream.